airflow/executors/celery_executor.py

```diff
@@ -41,6 +41,8 @@
 
 CELERY_SEND_ERR_MSG_HEADER = 'Error sending Celery task'
 
+SEND_TASK_TIMEOUT = conf.getint('celery', 'SEND_TASK_TIMEOUT')
```

Comment:
```suggestion
SEND_TASK_TIMEOUT = conf.getint('celery', 'CONNECTION_TIMEOUT')
```
You also wanna use timeout for fetching task states.